pydantic/_internal/_fields.py

```diff
@@ -229,6 +230,7 @@ def collect_fields(  # noqa: C901
             # 2. To avoid false positives in the NameError check above
             try:
                 delattr(cls, ann_name)
+                cls = cast('type[BaseModel]', cls)
```

Comment:
could you add a comment explaining how we know this cast is safe? I'm assuming it's related to the fact that `__pydantic_generic_x__` was present, but worth being clear